src/Log.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ * This file is part of OPUS. The software OPUS has been originally developed
+ * at the University of Stuttgart with funding from the German Research Net,
+ * the Federal Department of Higher Education and Research and the Ministry
+ * of Science, Research and the Arts of the State of Baden-Wuerttemberg.
+ *
+ * OPUS 4 is a complete rewrite of the original OPUS software and was developed
+ * by the Stuttgart University Library, the Library Service Center
+ * Baden-Wuerttemberg, the Cooperative Library Network Berlin-Brandenburg,
+ * the Saarland University and State Library, the Saxon State Library -
+ * Dresden State and University Library, the Bielefeld University Library and
+ * the University Library of Hamburg University of Technology with funding from
+ * the German Research Foundation and the European Regional Development Fund.
+ *
+ * LICENCE
+ * OPUS is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the Licence, or any later version.
+ * OPUS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details. You should have received a copy of the GNU General Public License
+ * along with OPUS; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ * @category    opus4-common
+ * @package     Opus\Log
+ * @author      Kaustabh Barman <barman@zib.de>
+ * @copyright   Copyright (c) 2020, OPUS 4 development team
+ * @license     http://www.gnu.org/licenses/gpl.html General Public License
+ */
+
+namespace Opus;
+
+use Opus\Log\LogService;
+
+class Log extends \Zend_Log
+{
+    public function __construct(\Zend_Log_Writer_Stream $writer = null)
+    {
+        parent::__construct($writer);
+    }
+
+    /**
+     * Eliminates all prior priority filters and sets a new one.
+     *
+     * @param String $priority
+     * @throws \Zend_Log_Exception
+     */
+    public function setPriority($priority)
+    {
+        if ($priority === null) {
+            $priority = LogService::DEFAULT_PRIORITY;
```

Comment:
This is a bit tricky, because LogService uses Log and Log uses LogService. Of course if you start using the priority functions in LogService, you still have the dependency in both directions. It will work, but is it good design? Do we need to support strings for priority/level in this class? Why are we supporting it in LogService? Do we need to have `setPriority` get a default value if the argument is null? How about removing the filter? That behavior should be explained in the long description (in a short note). That would also allow doing the DOI logging without a filter again. Hmm, just thinking.

Reply:
We may remove the support for string if this class will mainly be used in `createLogger()` of `LogService` or in similar classes. I will try to remove filters in case of null.

Reply:
We mostly support strings in LogService because that makes the configuration files more readable. If a developer sets the priority in the code we don't Need to be that nice. ;-)

Reply:
Done. Removed support for string. Also, not dependant on `LogService` anymore.